torch_audiomentations/utils/config.py

```diff
@@ -0,0 +1,126 @@
+import warnings
+from typing import Any, Dict, Mapping, Text, Union
+from pathlib import Path
+import torch_audiomentations
+from torch_audiomentations.core.transforms_interface import BaseWaveformTransform
+
+# TODO: remove this try/except/else once Compose is available
+# https://github.com/asteroid-team/torch-audiomentations/issues/23
+try:
+    from torch_audiomentations import Compose
+except ImportError:
+    COMPOSE_NOT_IMPLEMENTED = True
+else:
+    COMPOSE_NOT_IMPLEMENTED = False
+
+# TODO: define this elsewhere?
+# TODO: update when a new type of transform is added (e.g. BaseSpectrogramTransform?)
+# TODO: remove this if/else once Compose is available
+# https://github.com/asteroid-team/torch-audiomentations/issues/23
+if COMPOSE_NOT_IMPLEMENTED:
+    Transform = Union[BaseWaveformTransform]
+else:
+    Transform = Union[BaseWaveformTransform, Compose]
+
+
+def from_dict(config: Dict[Text, Dict[Text, Any]]) -> Transform:
+    """Instantiate a transform from a configuration dictionary.
+
+    `from_dict` can be used to instantiate a transform from its class name.
+    For instance, these two pieces of code are equivalent:
+
+    >>> from torch_audiomentations import TransformClassName
+    >>> transform = TransformClassName(param_name=param_value, ...)
+
+    >>> transform = from_dict({"TransformClassName": {"param_name": param_value, ...}}) 
+
+    Transforms composition is also supported:
+
+    >>> compose = from_dict({"FirstTransform": {"param": value},
+    ...                      "SecondTransform": {"param": value}})
```

Comment:
Hmm, a dict where the order of keys matters. Does that work well in Python 3.6? Perhaps it's better to use an OrderedDict to be on the safe side?

https://stackoverflow.com/a/39980744/2319697

Reply:
Agreed.

Reply:
How would that work with `from_yaml` for which `yaml.load` gives me a `dict` and not an `OrderedDict`? 
https://github.com/yaml/pyyaml/issues/110

There are two options I can think of:

- switch to a different syntax using a list (but this will make writing YAML more cumbersome)
- simpler but more involved: switch to Python 3.7+ for which Python insertion order is guaranteed (Python is now 3.9 already...)

Reply:
What would the syntax with list look like? I think I would prefer that if it means we can stay compatible with 3.6 (which lots of people are still using)

Reply:
Something like:
```yaml
- transform: FirstTransform
  params:
    param1: value1
    param2: value2
- transform: SecondTransform
  params:
    param3: value3
```

that converts to this Python list:

```python
[{'transform': 'FirstTransform',
  'params': {'param1': 'value1', 'param2': 'value2'}},
 {'transform': 'SecondTransform', 'params': {'param3': 'value3'}}]
```

Reply:
Looks acceptable to me. What do you think, @mpariente ?

Reply:
This looks ok, specially that Compose will probably have other arguments.